csrc/kernel_ir.cpp

```diff
@@ -61,12 +61,22 @@ ForLoop::ForLoop(
   NVF_ERROR(isIntegralType(index->dtype()));
   addInput(index);
   addInput(iter_domain);
-  if (start == nullptr && iter_domain->isThread()) {
-    start = NamedScalar::getParallelIndex(iter_domain->getParallelType());
+  if (start == nullptr) {
+    if (iter_domain->isThread() || iter_domain->isStream()) {
+      start = NamedScalar::getParallelIndex(iter_domain->getParallelType());
+    }
   }
   if (step == nullptr) {
     if (iter_domain->isThread()) {
       step = NamedScalar::getParallelDim(iter_domain->getParallelType());
+    } else if (iter_domain->isStream()) {
+      // `streamIdx` is a fixed kernel input (like `threadIdx`) that doesn't
+      // vary during execution of the kernel. Therefore, we emit a "trivial"
+      // for-loop of the form:
+      // ```
+      // for (nvfuser_index_t i0 = streamIdx; i0 < n_streams; i0 += n_streams)
+      // ```
+      step = iter_domain->extent();
```

Comment:
This probably doesn't matter, but do we consider where `iter_domain->extent() < n_streams`? That is, if multiple tensors are parallelized with streams and the parallelized IDs have different extents, is the number of streams the maximum of the extents of the parallelized IDs?

Reply:
So far, I've assumed they equal. I struggled to find a use case where they don't equal. 

How about threadIdx? Do we ever have extent < threadDim.x, and what does that mean?

Reply:
In the case of TID/BID, they are not guaranteed to be uniform across a fusion. IDs with different extents may use the same parallel type like TID, and we generate predicates like `threadIdx.x < id->extent()`. 

This is not very common, but generally we do not assume the parallel types are uniformly used, and GpuLower should support such patterns. 

Maybe padding? An iter domain with TIDx is padded by just 1, and then the padded iter domain is also parallelized with TIDx. The second ID is fine, but for the first ID a predicate is generated.

I think it's reasonable to assume the extent is always equal to the number of streams. It would be good to have some assertion.